src/estimagic/differentiation/derivatives.py

```diff
@@ -998,25 +934,36 @@ def _split_into_str_and_int(s):
     return str_part, int(int_part)
 
 
-def _collect_additional_info(return_info, steps, evals, updated_candidates, target):
+def _collect_additional_info(steps, evals, updated_candidates, target):
     """Combine additional information in dict if return_info is True."""
     info = {}
-    if return_info:
-        # save function evaluations to accessible data frame
-        if target == "first_derivative":
-            func_evals = _convert_evaluation_data_to_frame(steps, evals)
-            info["func_evals"] = func_evals
-        else:
-            one_step = _convert_evaluation_data_to_frame(steps, evals["one_step"])
-            info["func_evals_one_step"] = one_step
-            info["func_evals_two_step"] = None
-            info["func_evals_cross_step"] = None
-
-        if updated_candidates is not None:
-            # combine derivative candidates in accessible data frame
-            derivative_candidates = _convert_richardson_candidates_to_frame(
-                *updated_candidates
-            )
-            info["derivative_candidates"] = derivative_candidates
+    # save function evaluations to accessible data frame
+    if target == "first_derivative":
+        func_evals = _convert_evaluation_data_to_frame(steps, evals)
+        info["func_evals"] = func_evals
+    else:
+        one_step = _convert_evaluation_data_to_frame(steps, evals["one_step"])
+        info["func_evals_one_step"] = one_step
+        info["func_evals_two_step"] = None
+        info["func_evals_cross_step"] = None
+
+    if updated_candidates is not None:
+        # combine derivative candidates in accessible data frame
+        derivative_candidates = _convert_richardson_candidates_to_frame(
+            *updated_candidates
+        )
+        info["derivative_candidates"] = derivative_candidates
 
     return info
+
+
+def _is_exactly_nan(value):
+    return isinstance(value, float) and np.isnan(value)
+
+
+def _unflatten_if_ndarray(leaves, tree_def, registry):
```

Comment:
I guess if it's not a ndarray it's a scalar nan? If so, I would rename to `_unflatten_if_not_nan`. Partially because I find it more readable, partially because it should work for alternative array implementations such as jax arrays.